src/dbnode/storage/index/block.go

```diff
@@ -94,26 +162,45 @@ func NewBlock(
 	md namespace.Metadata,
 	opts Options,
 ) (Block, error) {
-	var (
-		blockSize = md.Options().IndexOptions().BlockSize()
-	)
+	docsPool := opts.DocumentArrayPool()
 
-	// FOLLOWUP(prateek): use this to track segments when we have multiple segments in a Block.
-	postingsOffset := postings.ID(0)
-	seg, err := mem.NewSegment(postingsOffset, opts.MemSegmentOptions())
+	foregroundCompactor, err := compaction.NewCompactor(docsPool,
+		documentArrayPoolCapacity,
+		opts.SegmentBuilderOptions(),
+		opts.FSTSegmentOptions())
```

Comment:
You need to set the registry cell size to zero here once we fix the underlying lib right?